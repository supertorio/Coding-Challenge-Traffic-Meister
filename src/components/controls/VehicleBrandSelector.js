import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {List as ImmutableList} from "immutable";

import {getAvailableVehicleBrands, getCurrentVehicleBrand} from "../../reducers/index-reducer";
import {changeVehicleBrandSelection} from "../../actions/meister-form-actions";
import BaseListSelector from "./BaseListSelector";

const VehicleBrandSelector = (props) => {
    return (
        <div className="Selector VehicleBrandSelector">
            <BaseListSelector label="Vehicle Brand"
                              options={props.brands}
                              value={props.selectedBrand}
                              onChange={props.changeVehicleBrandSelection} />
        </div>
    );
};

VehicleBrandSelector.propTypes = {
    brands: PropTypes.instanceOf(ImmutableList).isRequired,
    selectedBrand: PropTypes.string,
    changeVehicleBrandSelection: PropTypes.func,
};

const mapStateToProps = state => {
    return {
        brands: getAvailableVehicleBrands(state),
        selectedBrand: getCurrentVehicleBrand(state)
    };
};

const mapDispatchToProps = {
    changeVehicleBrandSelection
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(VehicleBrandSelector);
