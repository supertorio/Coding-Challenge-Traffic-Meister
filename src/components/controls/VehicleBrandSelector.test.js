import React from 'react';
import {VehicleBrandSelector} from './VehicleBrandSelector';
import {List} from 'immutable';

describe('<VehicleBrandSelector />', () => {

    let brands, selectedBrand;

    beforeEach(()=> {
        brands = new List(['Bugatti Veyron', 'Boeing 787 Dreamliner', 'Lamborghini HuracÃ¡n']);
        selectedBrand = brands.get[0];
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(
            <VehicleBrandSelector brands={brands} selectedBrand={selectedBrand} changeVehicleBrandSelection={()=> true} />,
        );
        expect(wrapper).toMatchSnapshot();
    });

});
