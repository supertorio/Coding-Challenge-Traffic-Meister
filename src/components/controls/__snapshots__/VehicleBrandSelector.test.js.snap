// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VehicleBrandSelector /> should match the snapshot 1`] = `ShallowWrapper {}`;
